src/main/java/com/redhat/coolstore/service/InventoryNotificationMDB.java

```diff
@@ -1,79 +1,48 @@
 package com.redhat.coolstore.service;
 
 import com.redhat.coolstore.model.Order;
-import com.redhat.coolstore.utils.Transformers;
-
+import io.quarkus.arc.Injectable;
+import io.quarkus.smallrye.reactivemessaging.api.ReactiveMessage;
+import io.smallrye.reactive.messaging.annotations.Channel;
+import io.smallrye.reactive.messaging.annotations.Incoming;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.jboss.logging.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
 import javax.inject.Inject;
-import javax.jms.*;
-import javax.naming.Context;
-import javax.naming.InitialContext;
-import javax.naming.NamingException;
-import javax.rmi.PortableRemoteObject;
-import java.util.Hashtable;
+import java.util.concurrent.CopyOnWriteArrayList;
 
-public class InventoryNotificationMDB implements MessageListener {
+@ApplicationScoped
+public class InventoryNotificationMDB {
 
     private static final int LOW_THRESHOLD = 50;
 
     @Inject
-    private CatalogService catalogService;
-
-    private final static String JNDI_FACTORY = "weblogic.jndi.WLInitialContextFactory";
-    private final static String JMS_FACTORY = "TCF";
-    private final static String TOPIC = "topic/orders";
-    private TopicConnection tcon;
-    private TopicSession tsession;
-    private TopicSubscriber tsubscriber;
-
-    public void onMessage(Message rcvMessage) {
-        TextMessage msg;
-        {
-            try {
-                System.out.println("received message inventory");
-                if (rcvMessage instanceof TextMessage) {
-                    msg = (TextMessage) rcvMessage;
-                    String orderStr = msg.getBody(String.class);
-                    Order order = Transformers.jsonToOrder(orderStr);
-                    order.getItemList().forEach(orderItem -> {
-                        int old_quantity = catalogService.getCatalogItemById(orderItem.getProductId()).getInventory().getQuantity();
-                        int new_quantity = old_quantity - orderItem.getQuantity();
-                        if (new_quantity < LOW_THRESHOLD) {
-                            System.out.println("Inventory for item " + orderItem.getProductId() + " is below threshold (" + LOW_THRESHOLD + "), contact supplier!");
-                        } else {
-                            orderItem.setQuantity(new_quantity);
-                        }
-                    });
-                }
+    CatalogService catalogService;
 
+    @ConfigProperty(name = "quarkus.smallrye.reactive-messaging.inventory-notifier")
+    String inventoryNotifierChannel;
 
-            } catch (JMSException jmse) {
-                System.err.println("An exception occurred: " + jmse.getMessage());
+    @Inject
+    @Channel("orderList")
+    ReactiveMessage<Order> orderList;
```

Comment:
It's coming up with stuff on its own instead of sticking to what's being told to do I think